pkg/events/watcher.go

```diff
@@ -76,7 +77,7 @@ type eventWatcher struct {
 	eventHandler EventHandler
 }
 
-func (w *eventWatcher) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+func (w *eventWatcher) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
```

Comment:
Now that `Reconcile` takes a context, can we pass it through to the events?

Reply:
do we know what the lifetime of the context is? currently we manage context outside of controller-runtime across our projects. probably would be smart to stage this as a separate change, and be clear about the lifetime (and other properties) of the context that we're using